pkgs/by-name/sw/swhkd/package.nix

```diff
@@ -0,0 +1,36 @@
+{lib, rustPlatform, fetchFromGitHub, pkg-config, pkgs }:
+
+  rustPlatform.buildRustPackage rec {
+    pname = "swhkd";
+    version = "1.2.1";
+
+    src = fetchFromGitHub {
+        owner = "waycrate";
+        repo = "swhkd";
+        rev = "refs/tags/${version}";
+        hash = "sha256-VQW01j2RxhLUx59LAopZEdA7TyZBsJrF1Ym3LumvFqA=";
+      };
+
+    nativeBuildInputs = with pkgs; [
+      pkg-config
+    ];
+
+    cargoHash = "sha256-NAVqwYJA0+X0dFC3PBaW+QJxvJtSgl4Y/VNfNO3jnLA=";
+
+    preBuild = ''
+      cargo generate-lockfile
+    '';
+
+    buildInputs = with pkgs; [
+      systemd
+      libgcc
+    ];
+
+    meta = with lib; {
+        description = "Wayland fork of sxhkd";
```

Comment:
for posterity: this is incorrect, swhkd is written in rust, sxhkd is written in C, they do not share a codebase, although they are meant to be compatible